Aanderaa 5990 Conductivity Sensor CBOR message

uma-kat opened the pull request:
Shortcut Ticket: [sc-228403 Create conductivity CBOR codec in bm_common_messages](https://app.shortcut.com/sofar/story/228403/create-conductivity-cbor-codec-in-bm-common-messages)

Created:
- `conductivity_data.msg` with these parameters
   ```
  float32 conductivity_ms_cm
  float32 temperature_deg_c
  float64 salinity_psu
  float64 water_density_kg_m3
  float64 sound_speed_m_s
  ```
- `conductivity_msg.h` 
- `conducticity_msg.cpp`
- test case `AanderaaConductivityTest` to `bm_common_messages_tests_ut.cpp`

matt001k reviewed (approved):
Nice work Uma!
As a heads up for the future, there are some nice helper functions in `bm_messages_helper.h` that could make this a bit easier to read/write :)

victorsowa12 commented on msg/conductivity_data.msg:
```diff
@@ -0,0 +1,8 @@
+import sensor_header
+
+sensor_header header
+float32 conductivity_ms_cm
+float32 temperature_deg_c
+float64 salinity_psu
+float64 water_density_kg_m3
+float64 sound_speed_m_s
```
any reason the conductivity and temp are float32, while the rest are float64?

uma-kat replied:
@victorsowa12 good catch. I misread something and assumed the precision was lower for conductivity and temperature. But it's the opposite! 🤦‍♀️ They have higher precision. I will change those to double as well.

victorsowa12 reviewed (approved):
Looks good to me! Had a question why some variables are float32(floats) and some are float64(doubles). I'm just curious if there is a reason based on sensor output.

As far as naming, (cc @matt001, @LarkspurRaven and @towynlin) how do we feel about naming these new files `aanderaa_conductivity_msg` and renaming the existing `aanderaa_data_msg` to `aanderaa_current_meter_msg`?

uma-kat commented:
@victorsowa12 good catch. I misread something and assumed the precision was lower for conductivity and temperature. But it's the opposite! 🤦‍♀️  They have higher precision. I will change those to double as well.

victorsowa12 commented:
Good eye @uma-kat! The current meter was the first sensor integrated so we likely didn't have our test infrastructure fully laid out yet. If you are up for it, could you add in that unit test?

uma-kat commented:
New changes since last review:
- Update data types for `conductivity_ms_cm` and `temperature_deg_c` to float64 (double)
- Renamed `conductivity_msg` to `aandera_conductivity_msg`
- Renamed `aanderaa_data_msg` to `aanderaa_current_meter_msg`
- Added unit test for the Aanderaa Current Meter -- `AanderaaCurrentMeterTest`
  - had to comment out "#include "FreeRTOS.h"" in `aanderaa_current_meter_msg.cpp` because it caused an error while running the build. Other sensor files don't seem to have this include.

matt001k commented on aanderaa_current_meter_msg.cpp:
```diff
@@ -1,8 +1,8 @@
-#include "aanderaa_data_msg.h"
-#include "FreeRTOS.h"
+#include "aanderaa_current_meter_msg.h"
+//#include "FreeRTOS.h"
```
This can be removed for cleanliness!

matt001k reviewed (approved):
Approval still stands!

victorsowa12 commented on CMakeLists.txt:
```diff
@@ -38,14 +38,15 @@ else()
     bm_rbr_pressure_difference_signal_msg.cpp
     bm_seapoint_turbidity_data_msg.cpp
     bm_soft_data_msg.cpp
+        aanderaa_conductivity_msg.cpp
```
nit: spacing
